File: examples/python/basic.py
"""
Tense Converter API - Basic Usage Example

This example demonstrates the basic usage of the Tense Converter API.
API Documentation: https://docs.apiverve.com/ref/tenseconverter
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/tenseconverter'

def call_tenseconverter_api():
    """
    Make a POST request to the Tense Converter API
    """
    try:
        # Request body
        request_body &#x3D; {
    &#x27;text&#x27;: &#x27;I am walking down the street, and I can see a dog&#x27;,
    &#x27;tense&#x27;: &#x27;past&#x27;
}

        headers = {
            'x-api-key': API_KEY,
            'Content-Type': 'application/json'
        }

        response = requests.post(API_URL, headers=headers, json=request_body)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Tense Converter API...\n')

    result = call_tenseconverter_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: npm/examples/basic.js
/**
 * Basic Example - Tense Converter API
 *
 * This example demonstrates how to use the Tense Converter API.
 * Make sure to set your API key in the .env file or replace '[YOUR_API_KEY]' below.
 */

require('dotenv').config();
const tenseconverterAPI = require('../index.js');

// Initialize the API client
const api = new tenseconverterAPI({
    api_key: process.env.API_KEY || '[YOUR_API_KEY]'
});

// Example query
var query = {
  "text": "I am walking down the street, and I can see a dog",
  "tense": "past"
};

// Make the API request using callback
console.log('Making request to Tense Converter API...\n');

api.execute(query, function (error, data) {
    if (error) {
        console.error('Error occurred:');
        if (error.error) {
            console.error('Message:', error.error);
            console.error('Status:', error.status);
        } else {
            console.error(JSON.stringify(error, null, 2));
        }
        return;
    }

    console.log('Response:');
    console.log(JSON.stringify(data, null, 2));
});
